### DISCUSSION TITLE
Option to hide/fold some assessments

### DISCUSSION THREAD

**Discussion by RaymondWHZ** [PrairieLearn/PrairieLearn#7564] [Ideas]
I want to add a new feature that allows users, primarily students, to optionally hide some assessments on their assessment list only to themselves. This may include:

* Assessments that they have done;
* Assessments that they do not plan to attempt;
* Previous attempts of quizzes that don't count;
* Any other reason that they don't want it to be shown to themselves.

The feature is going to allow the following:

* Conveniently hide specific assessments so that those assessments will be folded into some "hidden" sections. Those sections will be positioned near the original position of the assessment to avoid confusion. New assessments will be shown by default so that the student will not miss any assessment due to this feature;
* Quickly hide a whole section, all attempts of a quiz, or all attempts except the last one, but, as stated above, only existing items will be hidden so future assessments and attempts will still be shown if no specific action is taken.

This feature will be purely visual: no course data will be changed, no teachers or other students will be affected, and data can be stored in browser storage. Though, it would be better if the preference could also be saved to the student's account.

Some use cases of this feature:

* Hide all completed assessments, so that new-coming assessments will be very apparent;
* Hide a section that the student does not plan to complete to reduce distraction to focus on what needs to be done (potentially a non-credit section, depending on the individual's choice);
* Hide all previous attempts of quizzes to conveniently check the final score of each one.

Detailed UI design is still in progress, but I hope to discuss the feature a little bit first. Do you guys think this feature is relevant and appropriate? If it is, I will move forward to implement it.

**Comment by RaymondWHZ** [comment]
I have searched through the project and found a related issue #2086 as well as its pr #2185. I see its latest update is about three years ago, and it seems that no one is working on it right now. Moreover, the feature I'm requesting is more general and lightweight: it should allow students to click to hide ANY assessment instead of just attempts of a quiz/exam. It will also only change the appearance of the main page since the focus of this feature is not on improving the quiz/exam experience but allow the student to organize the assessment page overall.

**Comment by nwalters512** [comment]
Thanks for raising this for discussion! It's always useful to hear feedback and feature requests from students.

I'd like to dig a bit into the motivations for wanting this feature (a la https://xyproblem.info/). As a student, what's the reason you're interested in hiding assessments? I ask because I'm curious if there are deeper issues at play here. For instance, if you want this because it's difficult to tell when new assignments are available or keep track of when things are due, something like https://github.com/PrairieLearn/PrairieLearn/issues/152 may be a more generally useful solution. Or, if it's because the assessment page gets too cluttered with multiple instance assessments, https://github.com/PrairieLearn/PrairieLearn/pull/2185 (which I see you called out in your other comment) would be a great option.

**Reply by RaymondWHZ** [reply]
Thank you for your attention!

Indeed, my main issue is that the assessment page is too cluttered, not just because of multiple instance assessments but also regular assessments that I either have finished or have no intention to finish at all. Hence, I mainly hope that I can view the assessment page as a clear to-do list, which only shows items I have yet to complete so that I can focus on important upcoming items and get rid of items I no longer need to care about, so it is also a matter of focus. This is similar to Canvas, where completed assessments are folded in the dashboard view, but the hide action would be performed by students manually.

Hence, the main thing about the feature is to be customizable: the student has full control over what to show or hide instead of hiding multiple attempts by default.

There is a possible feature variant: give students the option to mark an entry as complete so that these entries are hidden by default and will be shown only if a "show complete" button is clicked.

What do you think of these?
